### DISCUSSION TITLE
Can I hide Bar cursor in inactive panes?

### DISCUSSION THREAD

**Discussion by eugenesvk** [wezterm/wezterm#1445] [Q&A]
The `Block` cursor has this nice effect — in inactive tabs/panes it turns into an outline (the filling color becomes the same as the background), so it's easy to distinguish between a focused cursor and an inactive one
However, the `Bar` cursor doesn't have this effect — its line color stays the same (which matches the behavior of the `Block` cursor, but then for the `Bar` cursor it doesn't work)

Is it possible to change the `cursor_border` color depending on whether the cursor is focused or not?

**Comment by wez** [comment]
The cursor color/styling is a bit complex so I want to be careful about making it more complex!

The block cursor is really the outline rectangle you see when the terminal isn't focused... when focused we have a special case and don't render a cursor graphic at all and instead just vary the cell colors.

When rendering a cursor graphic, it always uses the cursor border color for it.  (exception: when using the cursor as the bell target and the bell is ringing per: https://wezfurlong.org/wezterm/config/lua/config/visual_bell.html)

The colors are modified by [inactive_pane_hsb](https://wezfurlong.org/wezterm/config/appearance.html#styling-inactive-panes) but the default is perhaps a bit too subtle to notice on the likely choices of cursor border color: you might consider overriding the default to make the difference more visible?

I pushed d2e73e10cf7b94e407d6e89445020645f5066681 which causes wezterm to use the same cursor outline as block when the pane loses focus.  I'm not 100% sold on this change, but it has the advantage of being trivial and also of making the inactive cursor behavior consistent.

**Reply by eugenesvk** [reply]
> instead just vary the cell colors.

Oh, I see

>  [inactive_pane_hsb](https://wezfurlong.org/wezterm/config/appearance.html#styling-inactive-panes) ... you might consider overriding the default to make the difference more visible?

I've already done that, but since I'm using a lovely bright cursor color that stands out, to make the cursor really "hidden" I have to tone down the overall brightness a bit too much, so I lose more legibility in the inactive pane texts than I'd like 
Also, this doesn't affect the active tab when the window itself is out of focus, e.g., I have 4 panes, one active (bright), 3 inactive (dark) and then switch to a different window — instead of 4 dark "inactive" panes affected by `inactive_pane_hsb` I get the same 3 inactive ones and one bright "active" one

> I pushed [d2e73e1](https://github.com/wez/wezterm/commit/d2e73e10cf7b94e407d6e89445020645f5066681) which causes wezterm to use the same cursor outline as block when the pane loses focus. I'm not 100% sold on this change, but it has the advantage of being trivial and also of making the inactive cursor behavior consistent.

Ok, thanks, will check out later how that behaves

**Reply by eugenesvk** [reply]
> use the same cursor outline as block when the pane loses focus

this commit got me thinking: what do you think of introducing an "invisible" cursor (or a duplicate cursor that can be styled using separate fg/bg/border color variables and thus made invisible) and then using it in this commit? Or is it complicating it too much?

**Reply by wez** [reply]
we can make this do pretty much anything to be honest, so I'm not opposed to making changes.  I think the important thing is to have a good vision of what the end state should be so that result is clean and not hard to follow.

I've commented over in https://github.com/wez/wezterm/issues/1432#issuecomment-998936025 to summarize what I'm thinking of as an end state for cursor configuration.

**Comment by eugenesvk** [comment]
Ok, that summary is definitely more complicated that whatever I had in mind (I just wanted to match my editor's style, and width/height/color is all I need, I've used Neovide before and personally not a fan of that amount of fancy :) ); but if you believe there is enough demand for more complex modifications, then yeah, it might make more sense to complicate the lua config and give power to the users rather than incrementally reach the same hill with complicated internals
